lib/grp.gi

```diff
@@ -229,9 +233,13 @@ InstallMethod( IsPGroup,
         if ord = infinity then
           return false;
         elif 1 < ord then
-          UniteSet( s, Factors( ord ) );
-          if 1 < Length( s ) then
+          if not IsPrimePowerInt( ord ) then
             return false;
+          else
+            AddSet( s, Factors( ord )[1] );
+            if 1 < Length( s ) then
+              return false;
+            fi;
           fi;
```

Comment:
Ok, I can certainly do that. I tried to be as close to the original code as possible, but avoiding code dupication would be my preference, as well.